bin/varnishd/cache/cache_wrk.c

```diff
@@ -194,10 +194,8 @@ pool_getidleworker(struct pool *pp, enum task_prio prio)
 	}
 
 	if (pt == NULL) {
-		if (pp->nthr < cache_param->wthread_max) {
-			pp->dry++;
+		if (pp->nthr < cache_param->wthread_max)
 			AZ(pthread_cond_signal(&pp->herder_cond));
```

Comment:
I am not sure if waking up the herder makes any sense here without `dry`: We have not yet incremented `lqueue`, which happens in `Pool_Task()`, which is where I would think the signal should be posted.

Reply:
I agree, I also think we don't want to wake up the pool herder in the `Pool_Task_Arg` case.

Reply:
The lock is held all through the call to getidleworker and past the increment of lqueue, so the herder would not be able to run until after lqueue had been increment and would see it. It is a point for the case of having reached the queue limit, and would avoid a spurious herder wakeup in those cases. And it makes the code easier to read, so I agree with that change.